torch/_functorch/_aot_autograd/graph_capture.py

```diff
@@ -477,6 +477,8 @@ def aot_dispatch_autograd_graph(
     # After copying metadata, assign streams to gradient accumulation nodes
     assign_backward_streams(fx_g)
 
+    insert_backward_syncs(fx_g)
```

Comment:
Is there a reason why this has to be done on the joint graph? Trying to think if it makes partitioning harder.